### ISSUE TITLE
Is it possible to mannuly clear cache memory from gpu?

### ISSUE THREAD

**Issue by BrightXiaoHan** [OpenNMT/CTranslate2#824]
Hi, I deploy multiple (exactly three) model on a single gpu. I found the memory usage of each model will increase if I translate a large batch of data, and it will not decrease after translation. After a period of time, every model takes up a large chunk of memory. Is it because of the memory cache? Is it possible to mannuly clear it?

**Comment by guillaumekln** [comment]
Hi,

The additional memory is indeed because of the memory cache. Currently there is no way to manually clear it except by unloading the model, for example with `Translator.unload_model`.

We should update the code to improve this behavior which is mostly unexpected. It seems we can trigger a release of the memory cache by synchronizing the GPU. We could run this operation automatically after each batch translation so that the memory usage can decrease.

In the meantime, you can try the previous memory allocator with the environment variable [`CT2_CUDA_ALLOCATOR=cub_caching`](https://opennmt.net/CTranslate2/environment_variables.html#ct2-cuda-allocator). This allocator has a fixed cache size so the memory usage cannot continuously increase.

**Comment by guillaumekln** [comment]
Yes increasing the batch size will increase the cached memory. You can set the `max_batch_size` argument to limit the batch size for each translation.

**Comment by BrightXiaoHan** [comment]
Thanks a lot. My model was converted from fairseq using ctranslate version 2.8, and it seems that it not compatible with 2.19.0. I will try to re-convert it with 2.19.0 recently.

**Comment by BrightXiaoHan** [comment]
Thanks, version 2.19.0 works for me.
